### PR TITLE
Implemented elasticsearch validator

### PR DESCRIPTION
**Pull request by shrimetta**
**Problem Statement**
Validator for elasticsearch service

**Solution Overview**
Implemented validator for elasticsearch service

Signed-off-by: Shri Bhargav Metta <shri.metta@seagate.com>

### CODE REVIEW

File: py-utils/src/utils/validator/v_elasticsearch.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import errno
+
+from cortx.utils.validator.service import HttpService
+from cortx.utils.validator.error import VError
+
+
+class ElasticsearchV(HttpService):
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_elasticsearch.py:24]: This derivation is not required. HttpService can just be used for validation. 

```
class ElasticsearchV():
    ...

        if v_type == "service":
            host = args[0]
            port = args[1]
            HttpServer(host, port).validate_service_status()

```

File: py-utils/src/utils/validator/service.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        pass
+
+
+class HttpService(Service):
+    """
+    This class has validate_service_status implementation that makes http call
+    to check if service is up and running.
+    """
+
+    def __init__(self):
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/service.py:44]: init should take host, post, service as the parameters. They should not be default.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_elasticsearch.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        1. elasticsearch service localhost 9200
+        """
+
+        if not isinstance(args, list):
+            raise VError(errno.EINVAL, f"Invalid parameters {args}")
+
+        if len(args) < 2:
+            raise VError(errno.EINVAL, f"Insufficient parameters. {args}")
+
+        if v_type == "service":
```
Comment (ujjwalpl) [inline py-utils/src/utils/validator/v_elasticsearch.py:46]: Please make the similar change for consul as well so that code can be common for both.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/validator/v_elasticsearch.py (hunk 3 above)
Comment (shrimetta) [reply py-utils/src/utils/validator/v_elasticsearch.py:46]: I will raise separate PR for this.

File: py-utils/src/utils/validator/v_elasticsearch.py (hunk 1 above)
Comment (shrimetta) [reply py-utils/src/utils/validator/v_elasticsearch.py]: done

File: py-utils/src/utils/validator/service.py (hunk 2 above)
Comment (shrimetta) [reply py-utils/src/utils/validator/service.py]: done

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/validator/service.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        pass
+
+    def validate_service_status(self):
+        pass
+
+
+class HttpService(Service):
```
Comment (ajaysrivas) [inline py-utils/src/utils/validator/service.py:38]: I guess, the purpose of introducing this was to use it in both elasticsearch and consul but I do not see the code changes in Consul validator to use this. Do you need to make changes in Consul validator too?

Comment (shrimetta) [reply py-utils/src/utils/validator/service.py:38]: I will make changes for consul in separate checkin.

Comment (shrimetta) [reply py-utils/src/utils/validator/service.py:38]: Created issue #84 for the above

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**